diff -up gnome-keyring-2.28.2/daemon/gkr-daemon.c.bak gnome-keyring-2.28.2/daemon/gkr-daemon.c
--- gnome-keyring-2.28.2/daemon/gkr-daemon.c.bak	2009-12-14 15:38:06.000000000 +0100
+++ gnome-keyring-2.28.2/daemon/gkr-daemon.c	2012-02-27 17:25:58.000000000 +0100
@@ -220,16 +220,18 @@ static gboolean do_warning = TRUE;
  * locking for memory between threads
  */ 
 
+G_LOCK_DEFINE_STATIC (memory_mutex);
+
 void
 egg_memory_lock (void)
 {
-	/* The daemon uses cooperative threading, and doesn't need locking */
+	G_LOCK (memory_mutex);
 }
 
 void 
 egg_memory_unlock (void)
 {
-	/* The daemon uses cooperative threading, and doesn't need locking */
+	G_UNLOCK (memory_mutex);
 }
 
 void*
